### ISSUE TITLE
[TestFailures] EventSource_ExistsWithCorrectId fails on NativeAoT

### ISSUE THREAD

**Issue by rzikm** [dotnet/runtime#81824]
Happens deterministically on NativeAoT runs in System.Net.* tests (runtime-extra-platforms pipeline)

For example https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-8be2216db58a4d9e85/System.Net.Security.Tests/3/console.e0c8edc0.log?helixlogtype=result :

```
[FAIL] System.Net.Security.Tests.LoggingTest.EventSource_ExistsWithCorrectId
System.Resources.MissingManifestResourceException : Could not find the resource "FxResources.System.Net.Security.SR.resources" among the resources "" embedded in the assembly "System.Net.Security", nor among the resources in any satellite assemblies for the specified culture. Perhaps the resources were embedded with an incorrect name.
   at System.Resources.ManifestBasedResourceGroveler.HandleResourceStreamMissing(String) + 0x160
   at System.Resources.ManifestBasedResourceGroveler.GrovelForResourceSet(CultureInfo, Dictionary`2, Boolean, Boolean) + 0x1c4
   at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo, Boolean, Boolean) + 0x1bc
   at System.Resources.ResourceManager.GetString(String, CultureInfo) + 0x114
   at System.Diagnostics.Tracing.ManifestBuilder.WriteMessageAttrib(StringBuilder, String, String, String) + 0x70
   at System.Diagnostics.Tracing.ManifestBuilder.StartEvent(String, EventAttribute) + 0x138
   at System.Diagnostics.Tracing.EventSource.CreateManifestAndDescriptors(Type eventSourceType, String eventSourceDllName, EventSource source, EventManifestOptions flags) + 0xee4
   at System.Net.Security.Tests.LoggingTest.EventSource_ExistsWithCorrectId() + 0x104
   at System.Net.Security!<BaseAddress>+0xeb7a60
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0xf4
```

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

**Comment by rzikm** [comment]
Not sure about the labels. Feel free to edit.

**Comment by karelz** [comment]
@LakshanF @jkotas can you please help us route it? This is blocking CI significantly :(
Thanks!

**Comment by jkotas** [comment]
Introduced by #80896. Fixed by #81528.

The latest runtime-extra-platforms run does not have this failure: https://dev.azure.com/dnceng-public/public/_build/results?buildId=163126&view=ms.vss-test-web.build-test-results-tab

cc @MichalStrehovsky
